### PR TITLE
Add samtools.

### PR DESCRIPTION
**Pull request by DailyDreaming**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,25 +10,32 @@
...
+    print(f'Now running: {cmd}')
+    kwargs['shell'] = True
+    kwargs['stdout'] = subprocess.PIPE
+    kwargs['stderr'] = subprocess.PIPE
+    p = subprocess.Popen(cmd, **kwargs)
+    stdout, stderr = p.communicate()
+    if p.returncode:
+        print(f'\nstdout: {stdout}\n')
+        print(f'\nstderr: {stderr}\n\n')
+        raise subprocess.CalledProcessError(p.returncode, cmd, stdout, stderr)
```
Comment (xbrianh) [inline setup.py:23]: Please move the subprocess refactors into a separate PR.

Also, please use `run` instead of `Popen`. This package requires at least Python 3.7, and `run` provides stdout and stderr with a simpler API.

File: .gitignore [hunk 2]
```diff
@@ -11,6 +11,8 @@ __pycache__/
 /build/
 /dist/
 /.eggs/
+/venv/
+/v3nv/
```
Comment (xbrianh) [inline .gitignore:15]: Please move into separate PR.

**Comment by DailyDreaming** [comment]
@xbrianh Comments should be addressed.  :smiley:

File: Makefile [hunk 3]
```diff
@@ -16,7 +17,13 @@ tests:
...
+package_samtools: clean_samtools samtools.tar.bz2 htslib.tar.bz2 bcftools.tar.bz2
+
+samtools.tar.bz2:
+	git clone --depth 1 -b $(SAMTOOLS_VERSION) https://github.com/samtools/samtools.git
+	(cd samtools ; rm -rf .git ; autoheader && autoconf)
+	tar cjf samtools.tar.bz2 samtools
+	rm -rf samtools
 
 htslib.tar.bz2:
 	git clone --depth 1 -b xbrianh-readers-idx https://github.com/xbrianh/htslib
```
Comment (aofarrel) [inline Makefile:29]: I feel that it should be explicitly documented somewhere that this is a fork of htslib, as the htslib linked on the readme file is the samtools original.

Comment (DailyDreaming) [reply Makefile:29]: @aofarrel Hmmm... it's not generally a part of this PR (in practice, comments are restricted to the green and red diff lines because 1. it's easier to review smaller PRs and 2. this keeps you from following the slippery slope leading to refactoring entire files that might not have anything to do with the original PR)... but let me explain that I did this a bit differently because I felt the other forks should also be made off of their canonical repos respectively and pinned that way.

What should probably be done is to make all three cloned in the same way, but that's something that should be addressed in a separate PR.  I'd intended to do it anyway, so I'll go ahead and merge this, and follow up with that PR.

Also, readme options generally detail _how to use a product_ and leave out how individual dependencies are pinned and development details (except maybe at a very high level) as a general practice, and so linking the original repos, rather than the forks is still alright, since it's a user reference rather than an implementation reference.  This keeps things clear for the users, who may only interested in how to, say, use xsamtools to parse a file in a Google bucket.
